svgnative/src/SVGStringParser.cpp

```diff
@@ -29,54 +29,139 @@ void ArcToCurve(Path& path, float startX, float startY, float radiusX, float rad
 
 namespace SVGStringParser
 {
-using CharIt = std::string::const_iterator;
+class CharPos final {
```

Comment:
While I do like avoiding the `endPos` check, this can lead to a bigger performance degradation in a very performance critical path. We should run performance tests with huge path strings.